walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java

```diff
@@ -15,9 +15,117 @@
 package com.google.sps;
 
 import java.util.Collection;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.List;
+import java.util.HashSet;
+import java.util.Set;
 
 public final class FindMeetingQuery {
   public Collection<TimeRange> query(Collection<Event> events, MeetingRequest request) {
-    throw new UnsupportedOperationException("TODO: Implement this method.");
+      // meeting request has: name, duration, and attendees (List of Strings)
+      // event has: time range, attendees, and name
+
+      long duration = request.getDuration();
+
+       // meeting request exceeds one day
+      if (duration > TimeRange.WHOLE_DAY.duration()) {
+          return Arrays.asList();
+      }
+
+      // all attendees are free
+      if (events.isEmpty()) {
+          return Arrays.asList(TimeRange.WHOLE_DAY);
+      }
+
+      // there are no attendees
+      if (request.getAttendees().isEmpty()) {
+          return Arrays.asList(TimeRange.WHOLE_DAY);
+      }
+      
+
+      Collection<String> mandatory = request.getAttendees();
+      // Collection<String> optional = request.getOptionalAttendees();
+
+      
+      // retrieve all the times when at least one of the mandatory attendees 
+      // is busy
+      List<TimeRange> busy = new ArrayList<TimeRange> ();
+      for (Event e : events) {
```

Comment:
nit: event.   In general we want to avoid using single letter variable names, the most notable exception being when we're naming a variable for iterating through a collection by index e.g. "for (i=0; i < someSize; i++)"